### PR TITLE
RELATED: ONE-2818 reuse metric localIdentifier passed from AD

### PR DESCRIPTION
**Pull request by ivan-nejezchleb**

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/54/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/696/ : FAILURE in 39s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/3/ : FAILURE in 2m 18s (non-voting)

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/55/ : SUCCESS in 8s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/697/ : FAILURE in 30s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/4/ : FAILURE in 3m 21s (non-voting)

**Comment by vaclavbohac** [comment]
@ivan-nejezchleb [Q] Did you see #105? Sorting is getting completely revamped from the ground.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/56/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/698/ : SUCCESS in 1m 46s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/5/ : SUCCESS in 2m 14s (non-voting)

File: src/helpers/load.ts [hunk 1]
```diff
@@ -48,11 +48,22 @@ function handleExecutionError(reason: IError) {
...
 }
 
-function decorateMetrics(visObj: VisualizationObject.IVisualizationObject):
-    VisualizationObject.IVisualizationObject {
+function calculateLocalIdentifier(
+    item: VisualizationObject.IMeasure,
+    index: number
+): string {
+    // reuse identifier provided by AD
+    const localIdentifier = get(item, ['measure', 'localIdentifier'], '');
```
Comment (pbenes) [inline src/helpers/load.ts]: what about `return get(item, ['measure', 'localIdentifier'], `m${index + 1}`);

File: src/helpers/load.ts [hunk 2]
```diff
@@ -48,11 +48,22 @@ function handleExecutionError(reason: IError) {
...
+    return localIdentifier.length ? localIdentifier : `m${index + 1}`;
+}
+
+function decorateMetrics(
+    visObj: VisualizationObject.IVisualizationObject,
+): VisualizationObject.IVisualizationObject {
     const updatedVisObj = cloneDeep(visObj);
     updatedVisObj.content.buckets.measures = updatedVisObj.content.buckets.measures.map((measure, index) => {
-        measure.measure.generatedId = `m${index + 1}`;
+        measure.measure.localIdentifier = calculateLocalIdentifier(measure, index);
```
Comment (pbenes) [inline src/helpers/load.ts]: just a suggestion, what about `getLocalIdentifier`(calculate implies to me some "sophisticated" calculation :-) )

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/58/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/700/ : SUCCESS in 1m 46s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/7/ : SUCCESS in 2m 21s (non-voting)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/59/ : SUCCESS in 7s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/701/ : SUCCESS in 1m 05s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/8/ : SUCCESS in 2m 11s (non-voting)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-opensource-repository-check/62/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/704/ : SUCCESS in 2m 56s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/11/ : SUCCESS in 2m 15s (non-voting)

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-react-components-merge-compliance-check/74/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-el7/705/ : SUCCESS in 4m 37s
- https://ci.intgdc.com/job/gooddata-react-components-unit-tests-zuul-docker-el7/12/ : SUCCESS in 2m 14s (non-voting)

**Pull request merged**